include/irr/asset/IVirtualTexture.h

```diff
@@ -495,28 +587,34 @@ class IVirtualTexture : public core::IReferenceCounted
 
     virtual STextureData pack(const image_t* _img, const IImage::SSubresourceRange& _subres, ISampler::E_TEXTURE_CLAMP _wrapu, ISampler::E_TEXTURE_CLAMP _wrapv, ISampler::E_TEXTURE_BORDER_COLOR _borderColor) = 0;
 
-    virtual core::smart_refctd_ptr<image_view_t> createPageTableView() const = 0;
+    image_view_t* getPageTableView() const
+    {
+        if (!m_pageTableView)
+            m_pageTableView = createPageTableView();
+        return m_pageTableView.get();
+    }
 
-    image_t* const & getPageTable() const { return m_pageTable.get(); }
+    image_t* getPageTable() const { return m_pageTable.get(); }
     uint32_t getPageTableExtent_log2() const { return m_pgSzxy_log2; }
     uint32_t getPageExtent() const { return m_pgSzxy; }
     uint32_t getPageExtent_log2() const { return core::findLSB(m_pgSzxy); }
     uint32_t getTilePadding() const { return m_tilePadding; }
     const core::stack<uint32_t>& getFreePageTableLayersStack() const { return m_freePageTableLayerIDs; }
-    core::SRange<const uint32_t> getLayerToViewIndexMapping() const { return {m_layerToViewIndexMapping->begin(),m_layerToViewIndexMapping->end()}; }
+    core::SRange<const uint32_t> getLayerToViewIndexMapping() const { return {m_layerToViewIndexMapping.data(),m_layerToViewIndexMapping.data()+m_pageTable->getCreationParameters().arrayLayers}; }
     const auto& getResidentStorages() const { return m_storage; }
     typename SamplerArray::range_t getFloatViews() const  { return m_fsamplers.getViews(); }
     typename SamplerArray::range_t getIntViews() const { return m_isamplers.getViews(); }
     typename SamplerArray::range_t getUintViews() const { return m_usamplers.getViews(); }
 
     size_t getLayerToViewIndexLUTBytesize() const
     {
-        return m_layerToViewIndexMapping->size()*sizeof(uint32_t);
+        return m_pageTable->getCreationParameters().arrayLayers*sizeof(uint32_t);
     }
-    void writeLayerToViewIndexLUTContents(void* _dst) const
+    void writeLayerToViewIndexLUTContents(void* _dst, bool _updateFlag = true) const
     {
-        memcpy(_dst, m_layerToViewIndexMapping->data(), getLayerToViewIndexLUTBytesize());
-        m_layer2viewWasUpdatedSinceLastQuery = false;
+        memcpy(_dst, m_layerToViewIndexMapping.data(), getLayerToViewIndexLUTBytesize());
+        if (_updateFlag)
+            m_layer2viewWasUpdatedSinceLastQuery = false;
     }
     bool layerToViewIndexLUTWasUpdated() const
     {
```

Comment:
no point just return a constant reference plus take an argument if should reset the flag

Reply:
```cpp
inline const auto& getLayerToViewIndexMapping() const
{
    return m_layerToViewIndexMapping;
}
struct reset_update_t {};
const auto& getLayerToViewIndexMapping(reset_update_t reset_update)
{
    m_layer2viewWasUpdatedSinceLastQuery = false;
    return m_layerToViewIndexMapping;
}
```